sentry/src/routes/channel.rs

```diff
@@ -228,3 +234,169 @@ pub async fn create_validator_messages<A: Adapter + 'static>(
+    use primitives::config::configuration;
+    use primitives::BigNum;
+    use primitives::util::tests::discard_logger;
+
+    #[tokio::test]
+    async fn create_and_fetch_spendable() {
+        let database = DATABASE_POOL.get().await.expect("Should get a DB pool");
+        let redis = TESTS_POOL.get().await.expect("Should return Object");
+
+        let adapter_options = DummyAdapterOptions {
+            dummy_identity: IDS["leader"],
+            dummy_auth: IDS.clone(),
+            dummy_auth_tokens: AUTH.clone(),
+        };
+        let config = configuration("development", None).expect("Dev config should be available");
+        let dummy_adapter = DummyAdapter::init(adapter_options, &config);
+        setup_test_migrations(database.pool.clone())
+        .await
+        .expect("Migrations should succeed");
+        let mut channel = DUMMY_CAMPAIGN.channel.clone();
+        let token_address = Address::from_str("0x509ee0d083ddf8ac028f2a56731412edd63223b9").expect("should generate");
+        let token_info = config.token_address_whitelist.get(&token_address).expect("should retrieve address");
+        let precision: u8 = token_info.precision.into();
+        channel.token = token_address;
+        let deposit = Deposit {
+            total: BigNum::from(1000000000),
+            still_on_create2: BigNum::from(1000000),
+        };
+        dummy_adapter.add_deposit_call(channel.id(), ADDRESSES["creator"], deposit.clone());
+        let campaign_remaining = CampaignRemaining::new(redis.connection.clone());
+
+        let app = Application::new(
+            dummy_adapter,
+            config,
+            discard_logger(),
+            redis.connection.clone(),
+            database.pool.clone(),
+            campaign_remaining,
+        );
+        // Making sure spendable does not yet exist
+        let spendable = fetch_spendable(app.pool.clone(), &ADDRESSES["creator"], &channel.id()).await.expect("should return None");
+        assert!(spendable.is_none());
+
+        // Call create_spendable
+        let new_spendable = create_spendable_document(&app, &channel, &ADDRESSES["creator"]).await.expect("should create a new spendable");
+        assert_eq!(new_spendable.channel.id(), channel.id());
+
+        let total_as_unified_num = UnifiedNum::from_precision(deposit.total, precision).expect("should convert");
+        let still_on_create2_unified = UnifiedNum::from_precision(deposit.still_on_create2, precision).expect("should convert");
+        assert_eq!(new_spendable.deposit.total, total_as_unified_num);
+        assert_eq!(new_spendable.deposit.still_on_create2, still_on_create2_unified);
+        assert_eq!(new_spendable.spender, ADDRESSES["creator"]);
+
+        // Make sure spendable NOW exists
+        let spendable = fetch_spendable(database.pool.clone(), &ADDRESSES["creator"], &channel.id()).await.expect("should return a spendable");
+        assert!(spendable.is_some());
```

Comment:
- [ ] Continue the test case with scenario by updating the deposit and see if it gets reflected in `Spendable` after calling the `create_spendable_document`/`update_spendable_from_adapter`